src/mechanics/assembly_elasticity.hh

```diff
@@ -0,0 +1,695 @@
+                k++;
+            }
+        }
+        else if (bc_type == EqFields::bc_type_displacement_normal)
+        {
+            unsigned int k=0;
+            for (auto p : this->boundary_points(cell_side) ) {
+                for (unsigned int i=0; i<n_dofs_; i++)
+                    for (unsigned int j=0; j<n_dofs_; j++)
+                        local_matrix_[i*n_dofs_+j] += (eq_fields_->dirichlet_penalty(p) / side_measure) *
+                                arma::dot(vec_view_side_->value(i,k), fe_values_side_.normal_vector(k)) *
+                                arma::dot(vec_view_side_->value(j,k), fe_values_side_.normal_vector(k)) * fe_values_side_.JxW(k);
+                k++;
+            }
+        }
+
+        eq_data_->ls->mat_set_values(n_dofs_, dof_indices_.data(), n_dofs_, dof_indices_.data(), &(local_matrix_[0]));
+    }
+
+
+    /// Assembles between elements of different dimensions.
+    inline void neigbour_integral(DHCellAccessor cell_lower_dim, DHCellSide neighb_side) {
+    	if (dim == 1) return;
+        ASSERT_EQ_DBG(cell_lower_dim.dim(), dim-1).error("Dimension of element mismatch!");
+
+		cell_lower_dim.get_dof_indices(side_dof_indices_[0]);
+		ElementAccessor<3> cell_sub = cell_lower_dim.elm();
+		fe_values_sub_.reinit(cell_sub);
+
+		DHCellAccessor cell_higher_dim = eq_data_->dh_->cell_accessor_from_element( neighb_side.element().idx() );
+		cell_higher_dim.get_dof_indices(side_dof_indices_[1]);
+		fe_values_side_.reinit(neighb_side.side());
+
+		// Element id's for testing if they belong to local partition.
+		bool own_element_id[2];
+		own_element_id[0] = cell_lower_dim.is_own();
+		own_element_id[1] = cell_higher_dim.is_own();
+
+        for (unsigned int n=0; n<2; ++n)
+            for (unsigned int i=0; i<n_dofs_; i++)
+                for (unsigned int m=0; m<2; ++m)
+                    for (unsigned int j=0; j<n_dofs_; j++)
+                        local_matrix_ngh_[n][m][i*(n_dofs_)+j] = 0;
+
+        // set transmission conditions
+        unsigned int k=0;
+        for (auto p_high : this->coupling_points(neighb_side) )
+        {
+            auto p_low = p_high.lower_dim(cell_lower_dim);
+            arma::vec3 nv = fe_values_side_.normal_vector(k);
+
+            for (int n=0; n<2; n++)
+            {
+                if (!own_element_id[n]) continue;
+
+                for (unsigned int i=0; i<n_dofs_ngh_[n]; i++)
+                {
+                    arma::vec3 vi = (n==0) ? arma::zeros(3) : vec_view_side_->value(i,k);
+                    arma::vec3 vf = (n==1) ? arma::zeros(3) : vec_view_sub_->value(i,k);
+                    arma::mat33 gvft = (n==0) ? vec_view_sub_->grad(i,k) : arma::zeros(3,3);
+
+                    for (int m=0; m<2; m++)
+                        for (unsigned int j=0; j<n_dofs_ngh_[m]; j++) {
+                            arma::vec3 ui = (m==0) ? arma::zeros(3) : vec_view_side_->value(j,k);
```

Comment:
@jstebel We should avoid tests for individual dofs as it fails to generalize for different FE. And it is also slow as it breaks CPU pipelining.